src/components/AttachmentPicker/index.native.js

```diff
@@ -157,29 +156,20 @@ class AttachmentPicker extends Component {
             },
         ];
 
-        this.setResult = this.setResult.bind(this);
         this.close = this.close.bind(this);
-        this.completeAttachmentSelection = this.completeAttachmentSelection.bind(this);
+        this.pickAttachment = this.pickAttachment.bind(this);
     }
 
     /**
-     * Store the selected attachment mapped to an appropriate file interface
+     * Handles the image/document picker result and
+     * sends the selected attachment to the caller (parent component)
      *
      * @param {ImagePickerResponse|DocumentPickerResponse} attachment
      */
-    setResult(attachment) {
+    pickAttachment(attachment) {
         if (attachment && !attachment.didCancel && !attachment.error) {
             const result = getDataForUpload(attachment);
-            this.setState({result});
-        }
-    }
-
-    /**
-     * Triggers the `onPicked` callback with the selected attachment
-     */
-    completeAttachmentSelection() {
-        if (this.state.result) {
-            this.state.onPicked(this.state.result);
+            this.state.onPicked(result);
```

Comment:
Having `onPicked` stored in the state is not really a proper use of the state. The state should only store things that should affect the rendering of the component. Let's move `onPicked` to a class property instead. 

I know this  isn't exactly part of the issue you're working on, but I think cleaning up things like this in the code as we find them will keep the code quality high.